src/relay/backend/build_module.cc

```diff
@@ -0,0 +1,669 @@
+    return params_[name];
+  }
+
+  /*!
+   * \brief Set the parameters
+   *
+   * \param name name of parameter
+   * \param data_in input DLTensor
+   */
+  void _SetParams(const std::string& name, DLTensor* data_in) {
+    if (!params_.count(name)) {
+      std::vector<int64_t> shape(data_in->shape, data_in->shape + data_in->ndim);
+      params_[name] = tvm::runtime::NDArray::Empty(shape, data_in->dtype, {kDLCPU, 0});
+    }
+    params_[name].CopyFrom(data_in);
+  }
+
+  /*!
+   * \brief Set the optimization level
+   *
+   * \param level
+   */
+  void _SetOptLevel(char level) {
+    cfg_.opt_level = level;
+  }
+
+  /*!
+   * \brief type key
+   *
+   * \return const char*
+   */
+  const char* type_key() const final {
+    return "RelayBuildModule";
+  }
+
+  /*!
+   * \brief Build relay function for graph runtime
+   *
+   * \param func Relay Function
+   * \param target Target device
+   * \param target_host Host target device
+   */
+  void _Build(Function func,
+             const std::unordered_map<std::string, std::string>& targets,
+             const std::string& target_host) {
+    targets_ = targets;
+    target_host_ = target_host;
+    _BuildRelay(func, cfg_, params_);
+  }
+
+ protected:
+  /*!
+   * \brief bind params to function
+   *
+   * \param func Relay function
+   * \param params params dict
+   * \return relay::Function
+   */
+  relay::Function _bind_params_by_name(relay::Function func,
+                              const std::unordered_map<std::string, runtime::NDArray>& params) {
+    std::unordered_map<std::string, relay::Var> name_dict;
+    std::unordered_set<relay::Var, NodeHash, NodeEqual> repeat_var;
+    for (auto arg : func->params) {
+      const auto &name = arg->name_hint();
+      if (name_dict.count(name)) {
+        repeat_var.insert(arg);
+      } else {
+        name_dict[name] = arg;
+      }
+    }
+
+    std::unordered_map<relay::Var, Expr, NodeHash, NodeEqual> bind_dict;
+    for (auto &kv : params) {
+      if (name_dict.count(kv.first) == 0) {
+        continue;
+      }
+      auto arg = name_dict.at(kv.first);
+      if (repeat_var.count(arg)) {
+        LOG(FATAL) << "Multiple args in the function have name " << kv.first;
+      }
+      auto e = _CallPacked<Expr>("relay._make.Constant", kv.second);
+      bind_dict[arg] = e;
+    }
+    return _CallPacked("relay._expr.Bind", func, tvm::Map<relay::Var, Expr>(bind_dict));
+  }
+
+  /*!
+   * \brief Optimize Relay function
+   *
+   * \param func Input function
+   * \param target target device
+   * \param cfg Relay build config
+   * \param params params dict
+   * \return relay::Function
+   */
+  relay::Function _Optimize(relay::Function func,
+                            const std::unordered_map<std::string, std::string>& targets,
+                            const RelayBuildConfig& cfg,
+                            const std::unordered_map<std::string, runtime::NDArray>& params) {
+    if (params.size()) {
```

Comment:
```suggestion
    if (params.size() != 0) {
```
I assume _bind_params_by_name handle empty params. so I dont think you even need this conditional at all.

Reply:
It is consist to Python version